tree_builder.py

```diff
@@ -74,6 +76,15 @@ def fix_references(self, parent=None, root=None):
         for child in self.children.values():
             child.fix_references(self, root)
 
+    def clean_named_entities(self, parent=None, root=None):
+        if root is None:
```

Comment:
You're right. I was following the template of the previous recursive cleaning method and didn't really think about what it was doing. Fixed! I also think that if we continue to maintain this, for simplicity, it might be better to do this kind of preprocessing on the base text before tree creation.